monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -863,21 +969,65 @@ sealed abstract class Task[+A] extends Serializable {
   final def executeWithOptions(f: Options => Options): Task[A] =
     TaskExecuteWithOptions(this, f)
 
-  /** Returns a new task that will execute the source with autoCancelableRunLoops set to `true`
+  /** Returns a new task that is cancelable.
     *
-    * This will make `flatMap` driven loops to be auto-cancelable.
+    * Normally Monix Tasks have these characteristics:
     *
-    * Example:
+    * - `flatMap` chains are not cancelable by default
+    * - when creating [[Task.create async tasks]] the user has to specify explicit
+    *   cancellation logic
     *
+    * This operation returns a task that has [[Task.Options.autoCancelableRunLoops]]
+    * enabled upon evaluation, thus being equivalent with:
     * {{{
-    *   task.cancelable
+    *   task.executeWithOptions(_.enableAutoCancelableRunLoops)
     * }}}
     *
-    * This is equivalent to:
+    * What this does is two-fold:
+    *
+    * - `flatMap` chains become cancelable on async boundaries, which works in
+    *   combination with [[monix.execution.ExecutionModel.BatchedExecution BatchedExecution]]
+    *   that's enabled by default (injected by [[monix.execution.Scheduler Scheduler]],
+    *   but can also be changed with [[executeWithModel]])
+    * - even if the source task cannot be cancelled, upon completion the result
+    *   is not allowed to be streamed and the continuation is not allowed to execute
+    *
+    * For example this is a function that calculates the n-th Fibonacci element:
     * {{{
-    *   task.executeWithOptions(_.enableAutoCancelableRunLoops)
+    *   def fib(n: Int): Task[Long] = {
+    *     def loop(n: Int, a: Long, b: Long): Task[Long] =
+    *       Task.suspend {
+    *         if (n > 0)
+    *           loop(n - 1, b, a + b)
+    *         else
+    *           a
+    *       }
+    *
+    *     loop(n, 0, 1).cancelable
+    *   }
     * }}}
+    * 
+    * Normally this isn't cancelable and it might take a long time, but
+    * by calling `cancelable` on the result, we ensure that when cancellation
+    * is observed, at async boundaries, the loop will stop with the task
+    * becoming a non-terminating one.
+    * 
+    * This operation represents the opposite of [[uncancelable]]. And note
+    * that it works even for tasks that are uncancelable / atomic, because
+    * it blocks the rest of the `flatMap` loop from executing, functioning
+    * like a sort of cancellation boundary:
     *
+    * {{{
+    *   Task(println("Hello ..."))
+    *     .cancelable
+    *     .flatMap(_ => println("World!"))
```

Comment:
I thikn `println` needs to be wrapped in `Task`

Reply:
👍